src/System.Net.Security/tests/FunctionalTests/SslStreamSystemDefaultsTest.cs

```diff
@@ -53,14 +55,25 @@ public SslStreamSystemDefaultTest()
             var tasks = new Task[2];
             tasks[0] = AuthenticateClientAsync(serverHost, clientCertificates, checkCertificateRevocation: false, protocols: clientProtocols);
             tasks[1] = AuthenticateServerAsync(serverCertificate, clientCertificateRequired: true, checkCertificateRevocation: false, protocols: serverProtocols);
-            await await Task.WhenAny(tasks);
-            await Task.WhenAll(tasks);
 
-            if (PlatformDetection.IsWindows && PlatformDetection.WindowsVersion >= 10)
+            try
+            {
+                await Task.WhenAll(tasks);
```

Comment:
Use Task.WhenAllOrAnyFailedWithTimeout, so that the tests don't hang and timeout with networking test timeout limit..

Reply:
Good idea, I wasn't aware that such extension method had been added to `TestConfiguration` for the Security tests. Thanks.